vpn/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/AppTPVPNConflictDialog.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn.ui.tracker_activity
+
+import android.app.Dialog
+import android.os.Bundle
+import android.widget.Button
+import androidx.fragment.app.DialogFragment
+import com.duckduckgo.mobile.android.vpn.R
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class AppTPVPNConflictDialog : DialogFragment() {
+
+    interface Listener {
+        fun onDismissConflictDialog()
+        fun onOpenSettings()
+    }
+
+    val listener: Listener
+        get() {
+            return if (parentFragment is Listener) {
+                parentFragment as Listener
+            } else {
+                activity as Listener
+            }
+        }
```

Comment:
Why doing this in this way. It is a very weird factory pattern.
* we have an `instance()` factory method but we leave the listener out 
* It is not obvious that the parent needs to implement the `Listener` without looking at the implementation of the dialog
* it will crash if the parent doesn't implement the listener
* It is also weird that we provide an `instance` factory method but we leave a public constructor

I would change the signature of the constructor to
```kotlin
class AppTPVPNConflictDialog private constructor(private val listener: Listener): DialogFragment() {...}
```

And the factory method to 
```kotlin
    companion object {
        //...
        fun instance(listener: Listener): AppTPVPNConflictDialog {...}
    }
```